plugins/arStorageServicePlugin/modules/arStorageServicePlugin/actions/baseAction.class.php

```diff
@@ -0,0 +1,152 @@
+    catch (QubitApiBadRequestException $e)
+    {
+      $this->response->setStatusCode(400, $e->getMessage());
+    }
+    catch (Exception $e)
+    {
+      $this->response->setStatusCode(500);
+
+      throw $e;
+    }
+
+    return $view;
+  }
+
+  public function process($request)
+  {
+    $method = strtoupper($request->getMethod());
+    if (!method_exists($this, $method))
+    {
+      return $this->forward404();
+    }
+
+    // Define function callable
+    $fnCallable = array($this, $method);
+    $fnParamaters = array($request);
+
+    // Modern frameworks support application/json, Symfony1 is too old :)
+    // AngularJS doesn't use application/x-www-form-urlencoded
+    if (('PUT' == $method || 'POST' == $method) && 'application/json' == $request->getContentType())
+    {
+      $fnParamaters[] = json_decode($request->getContent());
+    }
+
+    // Load Qubit helper before calling it
+    ProjectConfiguration::getActive()->loadHelpers(array('Asset', 'Qubit'));
+
+    $result = call_user_func_array($fnCallable, $fnParamaters);
+
+    return $this->renderData($result);
+  }
+
+  public function renderData($data)
+  {
+    if ($data === 'CSV')
+    {
+      return sfView::NONE;
+    }
+
+    if ($data === sfView::NONE)
+    {
+      $this->response->setHeaderOnly(true);
+
+      $this->response->setStatusCode(204);
+
+      return sfView::NONE;
+    }
+
+    $options = 0;
+    if ($this->context->getConfiguration()->isDebug() && defined('JSON_PRETTY_PRINT'))
+    {
+      $options |= JSON_PRETTY_PRINT;
+    }
+
+    $this->response->setHttpHeader('Content-Type', 'application/json; charset=utf-8');
+
+    return $this->renderText(json_encode($data, $options));
+  }
+
+  public static function getFileFromStorageService($url)
+  {
+    $ch = curl_init();
+
+    curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'GET');
+
+    curl_setopt($ch, CURLOPT_URL,$url);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 500);
+    curl_setopt($ch, CURLOPT_WRITEFUNCTION, function($ch, $data) {
+        echo $data;
+        return strlen($data);
+    });
+
+    curl_setopt($ch, CURLOPT_HEADERFUNCTION, function($ch, $header) {
+        header($header);
+        return strlen($header);
+    });
+
+    curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1); // Storage service redirects
+    curl_setopt($ch, CURLOPT_FAILONERROR, true);
+    curl_setopt($ch, CURLOPT_HTTPHEADER, array(
+      sprintf('Authorization: ApiKey %s:%s', (string)QubitSetting::getByName('storage_service_username'), (string)QubitSetting::getByName('storage_service_api_key')),
+      'User-Agent: DRMC',
+    ));
+
+    curl_exec($ch);
+    $status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+    curl_close($ch);
+
+    return $status;
+  }
+
+  public static function getAipDownloadEnabled()
+  {
+    $setting = QubitSetting::getByName('download_aip_enabled');
+
+    return (null === $setting || boolval($setting->getValue(array('sourceCulture' => true))));
```

Comment:
I think this will return true if `null === $setting`, which I don't think is the intention?

Reply:
Fixed!